lib/grammers-session/src/message_box/mod.rs

```diff
@@ -344,8 +344,7 @@ impl MessageBox {
         } = match adaptor::adapt(updates, chat_hashes) {
             Ok(combined) => combined,
             Err(Gap) => {
-                self.try_begin_get_diff(Entry::AccountWide);
-                return Err(Gap);
+                panic!("adaptor::adapt should not trigger another gap");
```

Comment:
It's okay to trigger a gap when we process socket updates. What I meant by my previous comment was that we should panic if the gap is detected *while applying difference* (changing the propagation of the gap in the method that applies difference to an `expect`). But we should not panic here at the root of the error (because this code is also used to process socket updates which are expected to trigger gaps).